sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -44,8 +68,8 @@
   pubsub_pb2 = None
 
 
-__all__ = ['PubsubMessage', 'ReadMessagesFromPubSub', 'ReadStringsFromPubSub',
-           'WriteStringsToPubSub']
+__all__ = ['PubsubMessage', 'ReadMessagesFromPubSub', 'ReadPayloadsFromPubSub',
```

Comment:
Do we need 3 variations of ReadXFromPubSub transform? I would suggest considering making the underlying _ReadFromPubSub transform public, and letting users parametrize it to their needs.

Reply:
There was a previous discussions about this here: https://github.com/apache/beam/pull/3223 .
Java has [PubsubIO.readStrings()](https://github.com/apache/beam/blob/29859eb54d05b96a9db477e7bb04537510273bd2/examples/java/src/main/java/org/apache/beam/examples/complete/game/StatefulTeamScore.java#L126), [readMessages()](https://github.com/apache/beam/blob/29859eb54d05b96a9db477e7bb04537510273bd2/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java#L447), readMessagesWithAttributes(), etc.

Reply:
Thank you for the pointers.

I am trying to build a mental map of API available in both SDKs. You are proposing:

PTransforms:
- ReadMessagesFromPubSub: Outputs elements of type ``PubsubMessage``.
- ReadPayloadsFromPubSub: Outputs elements of type ``str``.
- ReadStringsFromPubSub: Outputs elements of type ``unicode``, decoded from UTF-8.

in Java there is
readMessages -> returns elements of type ``PubsubMessage.payload`
readMessagesWithAttributes -> returns both payload and attribute map
readStrings -> returns utf-8 decoded strings

There is also readProtos and readAvros.

Should we standardize on something like:
ReadStringsFromPubSub - strings
ReadMessagesFromPubSub - PubSubMessage (attributes being an argument to ReadMessagesFromPubSub, depending on that PubSubMessage will have the attributes or not.)

Thoughts?

Reply:
@lukecwik @charlesccychen can you comment on whether we actually need 'ReadStringsFromPubSub' for some reason that we missed here ?

Reply:
No, we don't need it. It was simply a tentative API choice at the time.

Reply:
After talking to @aaltay this morning, I'm going to reduce the API to just 1 transform that does everything.

Reply:
I maybe wrong. AFAIK, Dataflow implementation for reading only the payloads is faster that reading the whole message. I assume, we can still do this in DataflowRunner using transform overrides.